packages/strapi-design-system/src/DatePicker/DatePicker.tsx

```diff
@@ -549,12 +571,49 @@ const DatePickerTextInput = React.forwardRef<DatePickerTextInputElement, TextInp
             return;
           }
 
-          context.onTextValueChange(context.calendarDate.toString().split('-').reverse().join(separator));
+          context.onTextValueChange(formatter.format(context.calendarDate.toDate('UTC')));
           context.onValueChange(context.calendarDate);
         })}
         onChange={composeEventHandlers(props.onChange, (event) => {
           if (isPrintableCharacter(event.target.value)) {
-            const [day, month, year] = event.target.value.split(separator);
+            const inputByPart = event.target.value.split(separator);
+
+            /**
+             * by using the dateStructure to understand the localised order we split and organise the event.target.value
+             * to DD MM YYYY and return as an array.
+             */
+            const [day, month, year] = dateStructure
+              .map((part, index) => {
+                const value = inputByPart[index];
+
+                return {
+                  ...part,
+                  value,
+                };
+              })
+              /**
+               * TODO: This could probably be better done?
+               */
+              .sort((a, b) => {
+                if (a.type === 'year') {
+                  return 1;
+                }
+
+                if (b.type === 'year') {
+                  return -1;
+                }
+
+                if (a.type === 'month') {
+                  return 1;
+                }
+
+                if (b.type === 'month') {
+                  return -1;
+                }
+
+                return 0;
+              })
+              .map((part) => part.value);
```

Comment:
⚠️ Backender commenting on a frontend PR ⚠️ 

This is more of a question:
I am normally used to split logic and UI, so you have the domain defined elsewhere with how a datepicker should work, and you use those functionalities in the UI definitions. When I am reading the UI , I normally try to understand the functionality not the implementation, and having both in the same place makes it a bit more difficult to follow.

Do you think it could be useful to split domain logic here? Or it would be harder to track where each thing goes to?

Also, using a domain forces you to think things in an abstract way.

Reply:
EXCELLENT QUESTION. 

Typically we don't do that, because we don't test internals and if we export the internals e.g. the logic to convert any date format e.g. YY/DD/MM to DD/MM/YY – we then have to test that independently but realistically we're more satisfied with testing how a user interacts with things, the implementation is not as important because 9/10 at this level we're rarely sharing such niche code & it would make our suite more brittle than we deem valued.

Also from my perspective I believe in deep modules (big files) with a small as possible API surface so people understand whats for public consumption etc.

How do you feel about that?